data/inference/make_datasets/parse_utils.py

```diff
@@ -0,0 +1,27 @@
+import re
+
+
+def extract_scripts_from_codetext(codetext: str):
+    script_names = []
+    # 提供的文本内容，可能包含多个 [start of ... .py]
+    """
+    [end of README.rst]
+    [start of sklearn/compose/_target.py]
+    ... 文件内容 ...
+    [end of sklearn/compose/_target.py]
+    [start of another_module/example.py]
+    ... 文件内容 ...
+    [end of another_module/example.py]
+    """
+
+    # 使用正则表达式匹配所有 “[start of 任意字符.py]”
+    matches = re.findall(r"\[start of ([^\]]+\.py)\]", codetext)
```

Comment:
Does this function only aim to support parsing Python code? Because of the regex with a '.py' suffix.

Reply:
Yes, currently, it is only available for Python scripts.